core/src/tests/test_edit.cc

```diff
@@ -0,0 +1,161 @@
+/*
+   BAREOS® - Backup Archiving REcovery Open Sourced
+
+   Copyright (C) 2018-2020 Bareos GmbH & Co. KG
+
+   This program is Free Software; you can redistribute it and/or
+   modify it under the terms of version three of the GNU Affero General Public
+   License as published by the Free Software Foundation and included
+   in the file LICENSE.
+
+   This program is distributed in the hope that it will be useful, but
+   WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+   Affero General Public License for more details.
+
+   You should have received a copy of the GNU Affero General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+   02110-1301, USA.
+*/
+#if defined(HAVE_MINGW)
+#include "include/bareos.h"
+#include "gtest/gtest.h"
+#else
+#include "gtest/gtest.h"
+#include "include/bareos.h"
+#endif
+
+#include "lib/edit.h"
+
+TEST(edit, convert_number_to_siunits)
+{
+  ASSERT_STREQ(SizeAsSiPrefixFormat(1).c_str(), "1");
+  ASSERT_STREQ(SizeAsSiPrefixFormat(1024).c_str(), "1 k");
+  ASSERT_STREQ(SizeAsSiPrefixFormat(1048576).c_str(), "1 m");
+  ASSERT_STREQ(SizeAsSiPrefixFormat(1073741824).c_str(), "1 g");
+  ASSERT_STREQ(SizeAsSiPrefixFormat(1099511627776).c_str(), "1 t");
+  ASSERT_STREQ(SizeAsSiPrefixFormat(1125899906842624).c_str(), "1 p");
+  ASSERT_STREQ(SizeAsSiPrefixFormat(1152921504606846976).c_str(), "1 e");
+
+  ASSERT_STREQ(
+      SizeAsSiPrefixFormat(1152921504606846976 + 1125899906842624 +
+                           1099511627776 + 1073741824 + 1048576 + 1024 + 1)
+          .c_str(),
+      "1 e 1 p 1 t 1 g 1 m 1 k 1");
+}
+
+// kibibyte
+TEST(edit, convert_siunits_to_numbers)
+{
+  {
+    char str[] = "1 k";
+    uint64_t retvalue = 0;
+    size_to_uint64(str, &retvalue);
```

Comment:
changing `size_to_uint64(char*, uint64_t*)` into `size_to_uint64(const char*, uint64_t*)` should be rather trivial and would allow to pass constant strings.